modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -0,0 +1,671 @@
+			remove_menu_page( 'users.php' );
+			remove_submenu_page( 'users.php', 'users.php' );
+			remove_submenu_page( 'users.php', 'user-new.php' );
+			remove_submenu_page( 'users.php', 'profile.php' );
+			remove_submenu_page( 'users.php', 'grofiles-editor' );
+			remove_submenu_page( 'users.php', 'grofiles-user-settings' );
+
+			add_menu_page( esc_attr__( 'Users', 'jetpack' ), __( 'Users', 'jetpack' ), 'list_users', $users_slug, null, 'dashicons-admin-users', 70 );
+			add_submenu_page( $users_slug, esc_attr__( 'All People', 'jetpack' ), __( 'All People', 'jetpack' ), 'list_users', $users_slug, null, 5 );
+			add_submenu_page( $users_slug, esc_attr__( 'Add New', 'jetpack' ), __( 'Add New', 'jetpack' ), 'promote_users', $add_new_slug, null, 10 );
+			add_submenu_page( $users_slug, esc_attr__( 'My Profile', 'jetpack' ), __( 'My Profile', 'jetpack' ), 'read', $profile_slug, null, 15 );
+			add_submenu_page( $users_slug, esc_attr__( 'Account Settings', 'jetpack' ), __( 'Account Settings', 'jetpack' ), 'read', $account_slug, null, 20 );
+			$this->migrate_submenus( 'users.php', $users_slug );
+		} else {
+			remove_menu_page( 'profile.php' );
+			remove_submenu_page( 'profile.php', 'grofiles-editor' );
+			remove_submenu_page( 'profile.php', 'grofiles-user-settings' );
+
+			add_menu_page( esc_attr__( 'My Profile', 'jetpack' ), __( 'My Profile', 'jetpack' ), 'read', $profile_slug, null, 'dashicons-admin-users', 70 );
+			add_submenu_page( $profile_slug, esc_attr__( 'Account Settings', 'jetpack' ), __( 'Account Settings', 'jetpack' ), 'read', $account_slug, null, 5 );
+			$this->migrate_submenus( 'profile.php', $profile_slug );
+		}
+	}
+
+	/**
+	 * Adds Tools menu.
+	 *
+	 * @param string $domain  Site domain.
+	 * @param bool   $calypso Optional. Whether links should point to Calypso or wp-admin. Default true (Calypso).
+	 */
+	public function add_tools_menu( $domain, $calypso = true ) {
+		$admin_slug = 'tools.php';
+		$menu_slug  = $calypso ? 'https://wordpress.com/marketing/tools/' . $domain : $admin_slug;
+
+		add_submenu_page( $menu_slug, esc_attr__( 'Marketing', 'jetpack' ), __( 'Marketing', 'jetpack' ), 'manage_options', 'https://wordpress.com/marketing/tools/' . $domain, null, 5 );
+		add_submenu_page( $menu_slug, esc_attr__( 'Earn', 'jetpack' ), __( 'Earn', 'jetpack' ), 'manage_options', 'https://wordpress.com/earn/' . $domain, null, 10 );
+
+		if ( $calypso ) {
+			remove_menu_page( $admin_slug );
+			remove_submenu_page( $admin_slug, $admin_slug );
+			remove_submenu_page( $admin_slug, 'import.php' );
+			remove_submenu_page( $admin_slug, 'export.php' );
+			remove_submenu_page( $admin_slug, 'delete-blog' );
+
+			add_menu_page( esc_attr__( 'Tools', 'jetpack' ), __( 'Tools', 'jetpack' ), 'manage_options', $menu_slug, null, 'dashicons-admin-tools', 75 );
+			add_submenu_page( $menu_slug, esc_attr__( 'Import', 'jetpack' ), __( 'Import', 'jetpack' ), 'import', 'https://wordpress.com/import/' . $domain, null, 15 );
```

Comment:
Similarly to the above, do we want to redirect folks to the WordPress.com importers when they have installed a plugin with a custom importer that's not available on WordPress.com?

Reply:
The import/export items have an `if ( $calypso )` check preceding them, so they'll only get changed when the above mentioned toggle points to Calypso.